### ISSUE TITLE
Cannot set gradient stop color via a binding

### ISSUE THREAD

**Issue by mjfreelancing** [sthewissen/Xamarin.Forms.PancakeView#76]
There doesn't seem to be any support for this:

```
<pancake:PancakeView>
  <pancake:PancakeView.BackgroundGradientStops>
    <pancake:GradientStop Offset="0" Color="{Binding Source={x:Static themeModels:CurrentTheme.Instance}, Path=Primary}" />
  </pancake:PancakeView.BackgroundGradientStops>
</pancake:PancakeView>
```

Are there plans to make the properties bindable, or can I achieve this some other way (in XAML)?

**Comment by sthewissen** [comment]
I chose to make the entire GradientStop collection bindable instead, allowing you to define entire reusable gradients. I feel that for the sake of consistency that's cleaner than rewriting the same gradient over and over and having to make sure the colors are in the same order/the same color.

```
<Application xmlns="http://xamarin.com/schemas/2014/forms" xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml" x:Class="Thewissen.PancakeViewSample.App"
             xmlns:yummy="clr-namespace:Xamarin.Forms.PancakeView;assembly=Xamarin.Forms.PancakeView">
    <Application.Resources>
        <ResourceDictionary>
            <yummy:GradientStopCollection x:Key="Rainbow">
                <yummy:GradientStop Color="#f3855b" Offset="0" />
                <yummy:GradientStop Color="#fbcf93" Offset="0.2" />
                <yummy:GradientStop Color="#fbe960" Offset="0.4" />
                <yummy:GradientStop Color="#a0e67a" Offset="0.6" />
                <yummy:GradientStop Color="#33c6ee" Offset="0.8" />
                <yummy:GradientStop Color="#c652ba" Offset="1" />
            </yummy:GradientStopCollection>
        </ResourceDictionary>
    </Application.Resources>
</Application>
```

And in the page:
```
<yummy:PancakeView BackgroundGradientAngle="45" HorizontalOptions="FillAndExpand" 
HeightRequest="150" CornerRadius="10" 
BackgroundGradientStops="{StaticResource Rainbow}" />```

**Comment by mjfreelancing** [comment]
Makes sense. I can live with that, thanks.

**Comment by mjfreelancing** [comment]
I jumped in too soon, that still doesn't allow me to change my colours on the fly though - from what I can tell. I'm trying to change a background gradient based on theme colors in two different XAML files. From what I can tell I still cannot do this:
```
<pancake:GradientStopCollection x:Key="ThemedWeatherBackgroundGradients">
  <pancake:GradientStop Offset="0" Color="{Binding Source={x:Static themeModels:CurrentTheme.Instance}, Path=Primary}" />
  <pancake:GradientStop Offset="0.9" Color="{Binding Source={x:Static themeModels:CurrentTheme.Instance}, Path=Primary}" />
  <pancake:GradientStop Offset="1" Color="{Binding Source={x:Static themeModels:CurrentTheme.Instance}, Path=Surface}" />
</pancake:GradientStopCollection>
```

ie., I need to be able to change the colors dynamically. Is my only option to do this in the code behind?

**Comment by sthewissen** [comment]
I'm assuming that whatever color comes back from `"{Binding Source={x:Static themeModels:CurrentTheme.Instance}, Path=Primary}"` is a static or dynamic resource somewhere? Wouldn't you be able to make the gradient itself a static/dynamic resource in each theme and do the same trick?

**Comment by mjfreelancing** [comment]
Ok, may be I failed to mention one very important requirement - I need to be able to apply my colors as dynamic, not static.

CurrentTheme.Instance is a static instance of a class with a bunch of bindable properties. I had to do this to solve another problem where I needed the best of both worlds of DynamicResource and IValueConverter. My original problem is described here: https://forums.xamarin.com/discussion/comment/402359

TL;DR; - I need to be able to apply a color, but also apply an opacity or elevation - something that cannot be performed via {DynamicResource}

The class is nothing special: https://github.com/mjfreelancing/XF.Material.Themer/blob/master/XF.Material.Themer/Models/Themes/CurrentTheme.cs

But it allows me to do this sort of thing (dynamically):

```
<Style x:Key="TextMediumEmphasisStyle" TargetType="Label">
  <Setter Property="TextColor"
          Value="{Binding Source={x:Static themeModels:CurrentTheme.Instance},
            Path=OnSurface,
            Converter={StaticResource ColorOpacityConverter},
            ConverterParameter={x:Static local:Constants.MediumEmphasisOpacity}}" />
</Style>
```

The issue I have is that the theme can change, and therefore the updates need to be dynamic, not static. If it is static then I need to close/open the app to see the changes applied.

In your example, everything is static. Using some of my existing resources I can see this will compile:

```
<Color x:Key="HighEmphasis">#DEFFFFFF</Color>
...
<pancakeView:GradientStop Offset="1" Color="{StaticResource HighEmphasis}" />
```

But there's currently no way (that I can see) I can use DynamicResource or the binding approach I have used (the code example below shows why I need to use CurrentTheme.Instance in conjunction with a converter).

At the moment I have it working by doing this, but I'd rather do it in XAML if I can. May be I've been staring at it for too long ;-)

```
if (CurrentTheme.Instance.Theme == Theme.Light)
{
  WeatherInfo.BackgroundGradientStops[0].Color = CurrentTheme.Instance.Primary;
  WeatherInfo.BackgroundGradientStops[1].Color = CurrentTheme.Instance.Primary;
  WeatherInfo.BackgroundGradientStops[2].Color = CurrentTheme.Instance.Surface;
}
else
{
  var elevationConverter = new SurfaceElevationConverter();
  var brandedSurface = (Color)elevationConverter.Convert(CurrentTheme.Instance.BrandedSurface, typeof(Color), ElevationLevel.dp1, null);

  WeatherInfo.BackgroundGradientStops[0].Color = brandedSurface;
  WeatherInfo.BackgroundGradientStops[1].Color = brandedSurface;
  WeatherInfo.BackgroundGradientStops[2].Color = CurrentTheme.Instance.Background;
}
```

I'm guessing the original intention / expected use case would be for people to decide on their colors up front and they would never change, or they could be changed by switching XAML files - but even then, the colors in those files are fixed values. In my case, it would be like a user specifying colors at runtime and everything would magically update (by binding to a model).

Although my class is static, the values can change at runtime.

I hope this is making sense - if not I'll have to try and create a smaller project to demonstrate the problem I'm having. My current project is at https://github.com/mjfreelancing/XF.Material.Themer but it's probably too much to wade through (I understand busy scehdules).

**Comment by sthewissen** [comment]
I've started doing some work on bindable color properties for gradient stops.

**Comment by sthewissen** [comment]
All the work for this is done on the vnext branch ✅🎉

I don't have a specific release date for this yet. Been thinking of rewriting a lot of stuff, to be honest, to make it all a bit more consistent. I can pump out a beta release somewhere over the next few days though.
